addons/project/models/project_task.py

```diff
@@ -1110,13 +1110,11 @@ def _notify_by_email_prepare_rendering_context(self, message, msg_vals=False, mo
             message, msg_vals, model_description=model_description,
             force_email_company=force_email_company, force_email_lang=force_email_lang
         )
-        if self.date_deadline:
-            render_context['subtitles'].append(
-                _('Deadline: %s', self.date_deadline.strftime(get_lang(self.env).date_format)))
-        elif self.date_assign:
-            render_context['subtitles'].append(
-                _('Assigned On: %s', self.date_assign.strftime(get_lang(self.env).date_format)))
-        return render_context
+        if self.stage_id:
+            render_context['subtitles'].append(_('Stage: %s', self.stage_id.name))
+            return render_context
+        else:
```

Comment:
you could simply return render_context once (no need to return in the if, and in the else).

Reply:
@tde-banana-odoo  
I've made the necessary changes to the code. If there is any issues, please let me know.